File: backend/Spring/exchange/src/main/java/iset/project/exchange/entities/Livre.java
package iset.project.exchange.entities;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@Document
public class Livre {
    @Id
    private String id ;
    @Indexed(unique = true)
    private String titre ;
    private String auteur ;
    private String maison_edition;
    private Etat etat_livre ;
    private CategorieLivre categorie_livre;
    //private String uploaded_by;
    private Boolean donate ;
}

File: frontend/Echange-J-L-Angular/src/app/models/Livre.ts
import { CategorieLivre } from "./Categorie_livre";
import { Etat } from "./Etat";
export class Livre {

        id!: String;
        titre!: String;
        auteur!: String;
        maison_edition!: String;
        etat_livre!: Etat;
        categorie_livre!: CategorieLivre;
        uploaded_by!: String;

}
